### PR TITLE
feat(settings): Add telemetry opt-out UI + Switch

### LINKED ISSUE #9071: Build accounts telemetry opt-out flow for new-settings
We don't yet have UX, but this seems like it will be another row in the new settings page UI, with a checkbox or a button to toggle the opt-out state, and probably a link to some SUMO doc or legal doc with details about what opting out means.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-2931)

### LINKED ISSUE #9088: Define UX and copy for FxA metrics opt out
We will need to define

* the opt-out UX and copy for the FxA server
* any UX we might want to add to desktop or mobile Firefoxes to guide users from opting out of client telemetry to opting out of accounts metrics

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-2929)

### PR DESCRIPTION
**Pull request by LZoog**
## Because

* We want users to have the ability to opt-out of accounts-level telemetry if they wish to

## This pull request

* Adds the initial UI to Settings and Nav behind booleans only enabled in Storybook until the BE is finished
* Creates an a11y and l10n friendly on/off Switch component for this new section
* Updates storybook and tests

## Issue that this pull request solves

closes #9071 
closes #9088

---

This PR needs:
* Double confirmation this looks good from a copy standpoint, I was pinged towards the end of the day by someone legal conferred with and they'd like to review this
* I need check on RTL styles. This might need a _little_ more work on the switch in that regard

^Both of these can potentially be finished in #10732, but I'd ideally like to just get them done here

The new `Switch` is accessible and screenreader friendly. It's also l10n friendly, I tested some long words to make sure the on/off "label" (in quotes because it's not _actually_ the label) accounted for them.

Some screenshots:
image
image
image

German for "disabled":
image

telemetry-opt-out

### CODE REVIEW

File: packages/fxa-react/configs/tailwind.js [hunk 1]
```diff
@@ -62,10 +64,11 @@ module.exports = {
...
         '-52': '-13rem',
-        '3': '0.75rem',
-        '6': '1.25rem',
-        '10': '2.5rem',
-        '55': '13.75rem',
+        0.5: '0.125rem',
+        3: '0.75rem',
+        6: '1.25rem',
+        10: '2.5rem',
+        55: '13.75rem',
```
Comment (LZoog) [inline packages/fxa-react/configs/tailwind.js:71]: Prettier decided to make a few tweaks to sections of these files. Can investigate if anything looks incorrect.

File: packages/fxa-settings/src/components/DataCollection/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { DataCollection } from '.';
+
+storiesOf('Components/DataCollection', module).add('default', () => (
```
Comment (LZoog) [inline packages/fxa-settings/src/components/DataCollection/index.stories.tsx:9]: I would like to write stories in the [CSF format](https://storybook.js.org/docs/react/api/csf) like we are for stories in the auth-server but I kept getting eslint complaints and wasn't sure what the diff was. I'll file a maintenance task to make sure we can use the new format in Settings and maybe convert one or two files for reference.

File: packages/fxa-settings/src/components/Nav/index.test.tsx [hunk 3]
```diff
@@ -56,15 +56,34 @@ describe('Nav', () => {
...
+  });
+
+  it('conditionally renders DataCollection', () => {
+    render(
+      <AppContext.Provider value={{ account, config: getDefault() }}>
+        <Nav showDataCollection />
+      </AppContext.Provider>
+    );
+
+    expect(screen.getByTestId('nav-link-data-collection')).toHaveTextContent(
```
Comment (LZoog) [inline packages/fxa-settings/src/components/Nav/index.test.tsx:71]: I added some explicit checks to be sure users wouldn't see any of this stuff until FXA-4106 and scattered those TODO comments about to make them easy to find all relevant files for when working on that ticket. Though, I should probably add one to the "with telemetry opt-out" story, will do that with any other edits.

**Review by dannycoates — APPROVED** [review]
Looks great @LZoog!

**Comment by LZoog** [comment]
Per Slack, going to merge this now for other-work-rebasing purposes. I'll put in a follow up with any copy updates and RTL styles for the switch.

**Pull request merged**